spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -32,33 +31,45 @@
 
 import org.springframework.beans.factory.BeanCreationException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
-import org.springframework.cloud.stream.binder.BinderConfiguration;
-import org.springframework.cloud.stream.binder.BinderFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.stream.binder.BinderType;
 import org.springframework.cloud.stream.binder.BinderTypeRegistry;
 import org.springframework.cloud.stream.binder.DefaultBinderFactory;
 import org.springframework.cloud.stream.binder.DefaultBinderTypeRegistry;
+import org.springframework.cloud.stream.binding.CompositeMessageChannelConfigurer;
+import org.springframework.cloud.stream.binding.MessageChannelConfigurer;
+import org.springframework.cloud.stream.binding.MessageConverterConfigurer;
+import org.springframework.cloud.stream.binding.MessageSourceBindingTargetFactory;
+import org.springframework.cloud.stream.binding.SubscribableChannelBindingTargetFactory;
+import org.springframework.cloud.stream.converter.CompositeMessageConverterFactory;
 import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
 import org.springframework.context.annotation.Role;
 import org.springframework.core.io.Resource;
 import org.springframework.core.io.UrlResource;
 import org.springframework.core.io.support.PropertiesLoaderUtils;
-import org.springframework.util.Assert;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.handler.support.HandlerMethodArgumentResolversHolder;
+import org.springframework.messaging.handler.annotation.support.DefaultMessageHandlerMethodFactory;
+import org.springframework.messaging.handler.annotation.support.MessageHandlerMethodFactory;
 import org.springframework.util.ClassUtils;
 import org.springframework.util.StringUtils;
 
 /**
  * @author Marius Bogoevici
  * @author Ilayaperumal Gopinathan
  * @author Oleg Zhurakousky
+ * @author Soby Chacko
  */
 @Configuration
 @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+@EnableConfigurationProperties({ BindingServiceProperties.class })
+@Import({ContentTypeConfiguration.class})
 public class BinderFactoryConfiguration {
```

Comment:
This can no longer be named as `BinderFactoryConfiguration` since the factory configuration is now moved.

Reply:
Well, it's a public class, so we should deprecated with comment stating exactly what you said. Given that it's a configuration we can probably rename it in 2.2, - essentially giving a small warning window, but for now we stuck with it

Reply:
Right. Let's deprecate and make it `extends BinderFactoryAutoConfiguration` (or whatever you think is better name for this) with moving all the context over there.
However keep in mind that this one is already used in the SCT:
```
@AutoConfigureBefore(BindingServiceConfiguration.class)
@AutoConfigureAfter(SimpleTaskConfiguration.class)
public class TaskEventAutoConfiguration {
```

Reply:
Oh! Sorry, we talked here about different class.

So, yes, this one has to be renamed as well.
But as Oleg said: that's already a matter of separate issue/PR.

@ilayaperumalg ,  please, proceed and raise such an issue if you feel strong.

Thanks for your patience!